core/include/webview/detail/backends/cocoa_webkit.hh

```diff
@@ -234,22 +235,28 @@ protected:
       style =
           static_cast<NSWindowStyleMask>(style | NSWindowStyleMaskResizable);
     }
-    objc::msg_send<void>(m_window, "setStyleMask:"_sel, style);
-
-    if (hints == WEBVIEW_HINT_MIN) {
-      objc::msg_send<void>(m_window, "setContentMinSize:"_sel,
-                           CGSizeMake(width, height));
-    } else if (hints == WEBVIEW_HINT_MAX) {
-      objc::msg_send<void>(m_window, "setContentMaxSize:"_sel,
-                           CGSizeMake(width, height));
+    auto fn = [this, style, hints, width, height]() {
+      objc::msg_send<void>(m_window, "setStyleMask:"_sel, style);
+
+      if (hints == WEBVIEW_HINT_MIN) {
+        objc::msg_send<void>(m_window, "setContentMinSize:"_sel,
+                             CGSizeMake(width, height));
+      } else if (hints == WEBVIEW_HINT_MAX) {
+        objc::msg_send<void>(m_window, "setContentMaxSize:"_sel,
+                             CGSizeMake(width, height));
+      } else {
+        CGRect rect = objc::msg_send_stret<CGRect>(m_window, "frame"_sel);
+        objc::msg_send<void>(
+            m_window, "setFrame:display:animate:"_sel,
+            CGRectMake(rect.origin.x, rect.origin.y, width, height), YES, NO);
+      }
+      objc::msg_send<void>(m_window, "center"_sel);
+    };
+    if (is_main_thread()) {
+      fn();
     } else {
-      CGRect rect = objc::msg_send_stret<CGRect>(m_window, "frame"_sel);
-      objc::msg_send<void>(
-          m_window, "setFrame:display:animate:"_sel,
-          CGRectMake(rect.origin.x, rect.origin.y, width, height), YES, NO);
+      dispatch(fn);
```

Comment:
This change is needed to call `webview_set_size` from a child thread on MacOS via `dispatch`. See the [minimal reproduction](https://gist.github.com/citkane/6162e1fd345aa7f180fa0b9349fd07d2), which crashes without this patch.

The implementation here is a bit explicit, but I feel that it is warranted:
- We are better off not impacting the existing function behaviour when run on the `main` thread, ie. a direct synchronous call.
- Other Webview API calls that a user may use from a child thread are potentially also buggy, and this patch establishes a workable/flexible pattern.